src/GNDStk/Node/test/call.test.cpp

```diff
@@ -0,0 +1,124 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+
+// -----------------------------------------------------------------------------
+// Some child_ts
+// -----------------------------------------------------------------------------
+
+// temperature_t
+struct temperature_t {
+   // just the metadata, for this example
+   double value;
+   std::string unit;
+};
+
+inline void convert(const Node<> &n, temperature_t &temp)
+{
+   temp.value = n(meta::dvalue);
+   temp.unit  = n(meta::unit);
+}
+
+// isotope_t
+struct isotope_t {
+   std::string symbol;
+   int A;
+   const Node<> *nuclides;
+};
+
+inline void convert(const Node<> &n, isotope_t &iso)
+{
+   iso.symbol = n(meta::symbol);
+   iso.A = n(meta::A);
+   iso.nuclides = &n(child::nuclides);
+}
+
+
+
+// -----------------------------------------------------------------------------
+// Scenario
+// -----------------------------------------------------------------------------
+
+SCENARIO("Testing GNDStk Node operator()") {
+
+   // If I put these outside SCENARIO, the build system claims that they're
+   // multiply defined with respect to the definitions in child.test.cpp.
+   // Why on earth does the build system cram together that file and this one?
+   // Oh, but if I leave them out (because, apparently, the build system tries
+   // to bring them in from child.test.cpp!), then it complains that they aren't
+   // defined at all! OK, maybe putting them here, locally, will work. Hooray
+   // for confounding, convoluted and Byzantine building and testing systems
+   // that invariably make compiling and testing more complicated than they'd
+   // be if we had no such systems at all.
+
+   // keywords with find::one
+   auto temperature =
+      keyword.child<temperature_t,find::one>("temperature");
+   auto styles =
+      keyword.child<void,find::one>("styles");
+
+   // keywords with find::all
+   auto isotope =
+      keyword.child<isotope_t,find::all>("isotope");
+   auto isotope_node =
+      keyword.child<void,find::all>("isotope");
```

Comment:
I assume if the first template parameter is `void` it defaults to a `Node` type?

Reply:
Yes, that is correct. The `meta_t` and `child_t` objects have undergone a handful of tweaks to get them into what I think is the ideal form. For both, a first argument of `<void> `means: give us back the object in its native form in the tree. At present, that means `std::string` for metadata, `Node<something>` (something depends on the exact tree type!) for child nodes. If the internal format changes in the future, `<void>` will still mean: the native internal format, whatever that happens to be. (Not that I expect future changes to something this fundamental. Just sayin'.) There's more than this, however. The function overloads with `<void>` return *references* (const or non-const, as appropriate), because they're accessing the data in their native tree form. With `<non-void>`, only value-returns are possible (and then the functions don't need to be split into const and non-const because that's irrelevant with value returns). Note, then, that saying `meta_t<std::string>` and `meta_t<void>` are *not* the same; the first gives a value return, the second a reference return. Also, with the first, an optional conversion is available, analogous to `std::sort()`s optional ordering-predicate plugin: you can provide a conversion that modifies the string in some way before returning it. Whereas with `<void>`, you get a reference directly to the actual data.